proxy/http/ConnectingEntry.cc

```diff
@@ -0,0 +1,157 @@
+/** @file
+
+  Server side connection management.
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+ */
+
+#include "ConnectingEntry.h"
+#include "HttpSM.h"
+
+ConnectingEntry::~ConnectingEntry()
+{
+  if (_netvc_read_buffer != nullptr) {
+    free_MIOBuffer(_netvc_read_buffer);
+    _netvc_read_buffer = nullptr;
+  }
+}
+
+int
+ConnectingEntry::state_http_server_open(int event, void *data)
+{
+  Debug("http_connect", "entered inside ConnectingEntry::state_http_server_open");
+
+  switch (event) {
+  case NET_EVENT_OPEN: {
+    _netvc                 = static_cast<NetVConnection *>(data);
+    UnixNetVConnection *vc = static_cast<UnixNetVConnection *>(_netvc);
+    ink_release_assert(_pending_action == nullptr || _pending_action->continuation == vc->get_action()->continuation);
+    _pending_action = nullptr;
+    Debug("http_connect", "ConnectingEntrysetting handler for TCP handshake");
```

Comment:
The message implies code around this line is for TCP connections. Is it true?

I'm not going to say make this work with QUIC connections right now but I'm very sad if QUIC is completely out of your scope and we have to redesign it when we support QUIC to Origin. Wording that is not protocol-specific would help understanding what condition/state is expected.

More specifically, I'm wondering if TLS handshake has to be undone, because QUIC can't do TLS handshake separately.

Anyways, I don't want to make this change with H2-to-Origin at once. Connection pool should work without H2.

Reply:
That is just a working thing.  Can remove TCP.  Will need to test this in QUIC case.  There is a specific sequency of events that get sent as the TCP/TLS handshake works out.  Will probably be slightly different in the QUIC case.

In the meantime, this should be changed to more neutral language like "connection handshake" instead of "TCP handshake"

Reply:
Noticed that some Debug messages in HttpSM also refer to TCP.  They should also be cleaned up.